akka-http-core/src/main/scala/akka/http/impl/engine/http2/client/PersistentConnection.scala

```diff
@@ -185,11 +185,9 @@ private[http2] object PersistentConnection {
         }
         override def onPull(): Unit = responseIn.pull()
 
-        override def onUpstreamFinish(): Unit = {
+        override def onUpstreamFinish(): Unit =
           requestOut.complete()
-          responseIn.cancel()
-          completeStage()
-        }
```

Comment:
`PersistentConnection` should not cancel `responseIn` and complete the stage so eagerly.

Reply:
We should document here, that we here require "reasonable behavior" from downstream stages, i.e. that the downstream stage will eventually close all remaining inputs/outputs.

(A safer behavior would be to add another completion timeout here, but maybe we can leave that for now).

Reply:
> We should document here, that we here require "reasonable behavior" from downstream stages, i.e. that the downstream stage will eventually close all remaining inputs/outputs.

Will do.

> (A safer behavior would be to add another completion timeout here, but maybe we can leave that for now).

I think it's safe to leave that out for now.